copier/main.py

```diff
@@ -1105,8 +1105,15 @@ def _apply_update(self) -> None:  # noqa: C901
                         )
                         # Remove rejection witness
                         Path(f"{fname}.rej").unlink()
-                        # Store file name for marking it as unmerged after the loop
-                        conflicted.append(fname)
+                        # The 3-way merge might have resolved conflicts automatically,
+                        # so we need to check if the file contains conflict markers
+                        # before storing the file name for marking it as unmerged after the loop.
+                        with open(fname) as conflicts_candidate:
+                            if any(
+                                line.rstrip() in ("<<<<<<< before updating", ">>>>>>> after updating")
```

Comment:
suggestion: We might want to use global variables for the `before updating` and `after updating` strings, in case we change them in the future.